src/mkcalworker.cpp

```diff
@@ -642,16 +650,23 @@ bool mKCalWorker::removeCollections(const QList<QOrganizerCollectionId> &collect
         QStringList ids;
         QList<QOrganizerCollectionId> removedIds;
         QList<QPair<QOrganizerCollectionId, QOrganizerManager::Operation>> mods;
+        mKCal::Notebook::Ptr defaultNb = mStorage->defaultNotebook();
         int index = 0;
         for (const QOrganizerCollectionId &collectionId : collectionIds) {
             mKCal::Notebook::Ptr nb = mStorage->notebook(collectionId.localId());
             if (nb) {
-                if (!mStorage->deleteNotebook(nb)) {
+                // don't allow to remove a default collection
+                if (defaultNb == nb) {
```

Comment:
Notebook::Ptr are QSharedPointer<Notebook>. Testing equality between two QSharedPointer, I think, is testing if the pointing address in memory is the same. It would be safer to test the equality of the `uid()`. `defaultNotebook()` is not guaranteed to return a non null pointer, so add also a `!defaultNb` before the uid() equality test.